httpserver/rate_limiter.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <boost/circular_buffer.hpp>
+
+#include <chrono>
+#include <stdint.h>
+#include <string>
+#include <utility> // for std::pair
+
+/// https://en.wikipedia.org/wiki/Token_bucket
+
+class RateLimiter {
+  public:
+    // TODO: make those two constants command line parameters?
+    static const uint32_t BUCKET_SIZE = 5;
```

Comment:
Even if it is temporary, should probably start with something more reasonable, say 50 requests per second.

Reply:
changed